examples/crt/main.go

```diff
@@ -0,0 +1,291 @@
+// Copyright 2020 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build example
+
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"image"
+	"log"
+	"math"
+	"math/rand"
+	"time"
+	"image/color"
+	_ "image/png"
+
+	"golang.org/x/image/font"
+	"golang.org/x/image/font/opentype"
+
+	"github.com/hajimehoshi/ebiten/v2"
+	"github.com/hajimehoshi/ebiten/v2/examples/resources/fonts"
+	"github.com/hajimehoshi/ebiten/v2/inpututil"
+	"github.com/hajimehoshi/ebiten/v2/text"
+)
+
+// Game implements ebiten.Game interface.
+type Game struct {
+	Objects   []*Object
+	Offscreen *ebiten.Image
+	Time      float32
+	Score     int
+}
+
+// Object implements an object that the game loop modifies and the draw loop renders
+type Object struct {
+	Name          string
+	X, Y, Z, W, H int
+	VX, VY        float64
```

Comment:
Do these really need to be float64? Use int if possible.

Reply:
Float64 makes physics work properly.

Reply:
I'd avoid using float64 since it is pretty hard to treat errors.

Instead, I'd use a 'fixed-point arithmetic' by using ints that represent e.g., x16 scaled values. What do you think?

Reply:
ping

Reply:
Friendly ping :-)